src/main/java/org/takes/rq/RqLengthAware.java

```diff
@@ -58,17 +58,10 @@
      */
     public RqLengthAware(final Request req) {
         super(
-            new Request() {
-                @Override
-                public Iterable<String> head() throws IOException {
-                    return req.head();
-                }
-
-                @Override
-                public InputStream body() throws IOException {
-                    return RqLengthAware.cap(req);
-                }
-            }
+            new RequestOf(
+                req::head,
+                () -> RqLengthAware.cap(req)
```

Comment:
@fanifieiev same above.

Reply:
@fabriciofx I cannot use secondary constructor here.
![image](https://user-images.githubusercontent.com/3167940/69069241-25426580-09f4-11ea-9e75-02d195f1fe42.png)

Reply:
@fanifieiev as I said [here](https://github.com/yegor256/takes/pull/995#discussion_r347074093): *Of couse you can't use it in all places (e.g. beacause it can throws an Exception). So, in these places you can use the primary constructor...*